rotkehlchen/accounting/debugimporter/json.py

```diff
@@ -51,18 +52,23 @@ def import_history_debug(
         try:
             for event in debug_data['events']:
                 event_type = AccountingEventType.deserialize(event['accounting_event_type'])
-                if event_type == AccountingEventType.ASSET_MOVEMENT:
-                    events.append(AssetMovement.deserialize(event))
-                elif event_type == AccountingEventType.HISTORY_EVENT:
-                    events.append(HistoryEvent.deserialize(event))
-                elif event_type == AccountingEventType.LOAN:
-                    events.append(Loan.deserialize(event))
-                elif event_type == AccountingEventType.MARGIN_POSITION:
-                    events.append(MarginPosition.deserialize(event))
-                elif event_type == AccountingEventType.TRADE:
-                    events.append(Trade.deserialize(event))
-                elif event_type == AccountingEventType.STAKING:
-                    events.append(ValidatorDailyStats.deserialize(event))
+                match event_type:
+                    case AccountingEventType.TRADE:
+                        events.append(Trade.deserialize(event))
+                    case AccountingEventType.ASSET_MOVEMENT:
+                        events.append(AssetMovement.deserialize(event))
+                    case AccountingEventType.MARGIN_POSITION:
+                        events.append(MarginPosition.deserialize(event))
+                    case AccountingEventType.LOAN:
+                        events.append(Loan.deserialize(event))
+                    case AccountingEventType.STAKING:
+                        events.append(ValidatorDailyStats.deserialize(event))
+                    case AccountingEventType.HISTORY_EVENT:
+                        events.append(HistoryEvent.deserialize(event))
+                    case AccountingEventType.TRANSACTION_EVENT:
+                        events.append(EvmEvent.deserialize(event))
+                    case AccountingEventType.FEE | AccountingEventType.PREFORK_ACQUISITION:
```

Comment:
if a case is not covered does it warn you? And who warns? mypy? or the interpreter?

Reply:
I changed to match case because I recalled the interpreter warning me but seems that if the optinion is missing it becomes a no-op as per the docs and mypy doesn't detect it properly yet https://github.com/python/mypy/issues/13597